### PR TITLE
Add support for allowlisting hosts of images.

### PR DESCRIPTION
**Pull request by tmathews**
This commit allows us to permit certain domain hosts to use our service
so that it does not get abused by third parties. In the future I would
like to add * support, e.g. *.monstercat.com.

### CODE REVIEW

**Review by bryancua — APPROVED** [review]

File: server.go [hunk 1]
```diff
@@ -104,6 +53,11 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	opts.Prefix = "resized"
+
+	if !s.HostPermitted(opts.URL.Host) {
```
Comment (DerrickGold) [inline server.go:57]: So this only checks if someone is using it to resize monstercat assets and not where the call is being made from?

Comment (tmathews) [reply server.go:57]: Yep, we don't want to enable the functionality for other peoples data.

Comment (tmathews) [reply server.go:57]: You can't rely on the referrer header, but we could add a check there too.

Comment (DerrickGold) [reply server.go:57]: I think the best way to ensure that only we can call it and no one is through the use of service credentials or making this available within a VPC/VPN.

Comment (bryancua) [reply server.go:57]: in that case, we would need to put the asset-delivery behind our backend rather than in front.

Comment (DerrickGold) [reply server.go:57]: Also this only resizes images that we're already hosting on a google bucket right? So no body would be able to use it for their own images anyway?

**Pull request merged**